### PR TITLE
db: Cleanup obsolete tables

### PR DESCRIPTION
**Pull request by davidjumani**
### Description

Removes obsolete tables and code from the DB

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

### How Has This Been Tested?

TODO

### CODE REVIEW

File: engine/schema/src/main/resources/META-INF/db/schema-41510to41600.sql [hunk 1]
```diff
@@ -300,6 +300,21 @@ from
...
+
+DROP TABLE IF EXISTS `cloud`.`external_bigswitch_vns_devices`;
+DROP TABLE IF EXISTS `cloud`.`s3`;
+DROP TABLE IF EXISTS `cloud`.`swift`;
+DROP TABLE IF EXISTS `cloud`.`template_s3_ref`;
+DROP TABLE IF EXISTS `cloud`.`template_swift_ref `;
+DROP TABLE IF EXISTS `cloud`.`template_ovf_properties`;
+DROP TABLE IF EXISTS `cloud`.`op_host_upgrade`;
+DROP TABLE IF EXISTS `cloud`.`stack_maid `;
+DROP TABLE IF EXISTS `cloud`.`template_swift_ref `;
```
Comment (sureshanaparti) [inline engine/schema/src/main/resources/META-INF/db/schema-41510to41600.sql:315]: @davidjumani _template_swift_ref_ table already removed above

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Comment by davidjumani** [comment]
@blueorangutan test keepEnv

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

File: engine/schema/src/main/java/com/cloud/storage/dao/VMTemplateDaoImpl.java [hunk 2]
```diff
@@ -580,21 +524,6 @@ public void saveDetails(VMTemplateVO tmpl) {
...
-     * " GROUP BY t.id, h.data_center_id "; } if ((templateFilter ==
-     * TemplateFilter.featured) || (templateFilter == TemplateFilter.community))
-     * { dataCenterJoin =
-     * " INNER JOIN data_center dc on (h.data_center_id = dc.id)"; }
-     *
-     * if (zoneType != null) { dataCenterJoin =
-     * " INNER JOIN template_host_ref thr on (t.id = thr.template_id) INNER JOIN host h on (thr.host_id = h.id)"
-     * ; dataCenterJoin +=
-     * " INNER JOIN data_center dc on (h.data_center_id = dc.id)"; }
-     *
```
Comment (sureshanaparti) [inline engine/schema/src/main/java/com/cloud/storage/dao/VMTemplateDaoImpl.java:597]: this comment can be completely removed ?

File: engine/schema/src/main/java/com/cloud/storage/dao/VMTemplatePoolDaoImpl.java [hunk 3]
```diff
@@ -62,7 +62,7 @@
     protected final SearchBuilder<VMTemplateStoragePoolVO> updateStateSearch;
     protected final SearchBuilder<VMTemplateStoragePoolVO> templatePathSearch;
 
-    protected static final String UPDATE_TEMPLATE_HOST_REF = "UPDATE template_spool_ref SET download_state = ?, download_pct= ?, last_updated = ? "
+    protected static final String UPDATE_SPOOL_HOST_REF = "UPDATE template_spool_ref SET download_state = ?, download_pct= ?, last_updated = ? "
```
Comment (sureshanaparti) [inline engine/schema/src/main/java/com/cloud/storage/dao/VMTemplatePoolDaoImpl.java:65]:
```suggestion
    protected static final String UPDATE_TEMPLATE_SPOOL_HOST_REF = "UPDATE template_spool_ref SET download_state = ?, download_pct= ?, last_updated = ? "
```

**Review by sureshanaparti — APPROVED** [review]
cleanup code LGTM

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by borisstoyanov** [comment]
@blueorangutan test matrix

**Review by borisstoyanov — COMMENTED** [review]
@davidjumani @sureshanaparti there are quite a few marvin tests with this one. VMWare are coming with a lots of errors, xen also failing, can we check if they are related to the DB changes? 

VMware still running:
```
Intermittent failure detected: /marvin/tests/smoke/test_accounts.py
Intermittent failure detected: /marvin/tests/smoke/test_affinity_groups_projects.py
Intermittent failure detected: /marvin/tests/smoke/test_affinity_groups.py
Intermittent failure detected: /marvin/tests/smoke/test_async_job.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vgpu_enabled_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_extra_config_data.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_iso.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_root_resize.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vms_with_varied_deploymentplanners.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_with_userdata.py
Intermittent failure detected: /marvin/tests/smoke/test_diagnostics.py
Intermittent failure detected: /marvin/tests/smoke/test_domain_service_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_domain_vpc_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_clusters.py
Intermittent failure detected: /marvin/tests/smoke/test_list_ids_parameter.py
Intermittent failure detected: /marvin/tests/smoke/test_loadbalance.py
Intermittent failure detected: /marvin/tests/smoke/test_metrics_api.py
Intermittent failure detected: /marvin/tests/smoke/test_migration.py
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_nested_virtualization.py
Intermittent failure detected: /marvin/tests/smoke/test_network_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_network.py
Intermittent failure detected: /marvin/tests/smoke/test_nic_adapter_type.py
```

**Review by borisstoyanov — COMMENTED** [review]
@davidjumani @sureshanaparti there are quite a few marvin tests with this one. VMWare are coming with a lots of errors, xen also failing, can we check if they are related to the DB changes? 

suddenly this requires a guest_os 
```
get_test_template\n    template = Template.register
errorCode: 431, errorText:Please provide a guest OS type
```
VMware still running:
```
Intermittent failure detected: /marvin/tests/smoke/test_accounts.py
Intermittent failure detected: /marvin/tests/smoke/test_affinity_groups_projects.py
Intermittent failure detected: /marvin/tests/smoke/test_affinity_groups.py
Intermittent failure detected: /marvin/tests/smoke/test_async_job.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vgpu_enabled_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_extra_config_data.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_iso.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_root_resize.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vms_with_varied_deploymentplanners.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_with_userdata.py
Intermittent failure detected: /marvin/tests/smoke/test_diagnostics.py
Intermittent failure detected: /marvin/tests/smoke/test_domain_service_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_domain_vpc_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_clusters.py
Intermittent failure detected: /marvin/tests/smoke/test_list_ids_parameter.py
Intermittent failure detected: /marvin/tests/smoke/test_loadbalance.py
Intermittent failure detected: /marvin/tests/smoke/test_metrics_api.py
Intermittent failure detected: /marvin/tests/smoke/test_migration.py
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_nested_virtualization.py
Intermittent failure detected: /marvin/tests/smoke/test_network_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_network.py
Intermittent failure detected: /marvin/tests/smoke/test_nic_adapter_type.py
```

**Comment by sureshanaparti** [comment]
@borisstoyanov i think, these obsolete tables are not referred in these tests. @davidjumani can you re-confirm.

**Comment by davidjumani** [comment]
@borisstoyanov the obsolete tables aren't referenced in the tests. The intermittent failures are probably network / env related

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test matrix

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test centos7 vmware65

**Comment by davidjumani** [comment]
@blueorangutan test centos7 vmware65

**Review by borisstoyanov — APPROVED** [review]
LGTM, did an upgrade and the tables are properly cleaned. Smoke tests does not show any failures related to the changes. The latest failures are also present in other PRs and the previous were related to problems in our lab.

**Comment by nvazquez** [comment]
Hi @davidjumani @borisstoyanov is this PR ready for review?

**Comment by andrijapanicsb** [comment]
@pdion891 can you please review in regards to S3/swift table removals and related code - I would like to make sure this doesn't break thing for you guys, as I know you've been using S3/swift

cc @swill

**Pull request merged**